eng/tools/rush-runner/src/changed.js

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { existsSync } from "node:fs";
+import { readdir, readFile } from "node:fs/promises";
+import { join } from "node:path";
+
+// @ts-check
+
+/**
+ *
+ * @param {string} packageInfoPath - directory that store package info json files
+ * @param {string} changeInfoPath - path to the json file that store changed files (diff.json)
+ * @returns {{changedPackages: Set<string>, diff: { changedFiles: string[], changedServices: string[] }}}
+ */
+export async function getChangedInfo(packageInfoPath, changeInfoPath) {
+  const result = { packages: {} };
+  if (!existsSync(packageInfoPath) || !existsSync(changeInfoPath)) {
+    console.warn(
+      `Package info path (${packageInfoPath}) or change info path (${changeInfoPath}) is not specified or does not exist!`,
+    );
+    return result;
+  }
+
+  const changedPackages = new Set();
+  const dirPackageMap = new Map();
+  const files = await readdir(packageInfoPath);
+  for (const file of files) {
+    const content = await readFile(join(packageInfoPath, file), "utf-8");
+    const json = JSON.parse(content);
+    result["packages"][json["Name"]] = json;
+
+    // TODO: use IncludedForValidation instead when issue is resolved
+    dirPackageMap[json["DirectoryPath"]] = json["Name"];
+    // if (json["IncludedForValidation"]) {
```

Comment:
currently `IncludedForValidation` is not set correctly. all `false` in the PackageInfo json files.

Reply:
This is due to the fact that the PR diff for this PR has changes in `eng/`, and triggerPath added packages are treated as direct.